ext/nanopb/nanopb.lb

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021, Lucas Moesch
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+import os
+import subprocess
+import tempfile
+
+from pathlib import Path
+
+def init(module):
+    module.name = ":nanopb"
+    module.description = """
+# Nanopb - Protocol Buffers for Embedded Systems
+
+Nanopb is a small code-size Protocol Buffers implementation in ansi C. It is especially suitable 
+for use in microcontrollers, but fits any memory restricted system.
+"""
+
+def prepare(module, options):
+    module.add_option(
+    PathOption(name="protofile.path", empty_ok=False, absolute=True, description=""))

```

Comment:
```suggestion
    PathOption(name="protofile.path", empty_ok=False, absolute=True, description="", default=""))
```
Otherwise lbuild requires an option input without default.

Reply:
I'm not sure if a default should be provided here. With a default path it it possible to run a nanopb build with no source file to build.

Reply:
It's just that if you don't default the option, lbuild will cannot proceed if the module is selected (see the doc build job in CI).
I would make the behavior so that when the path is empty, everything is still copied, but the `{{ proto_source }}` is just missing from the SConstruct, for the user to fill it in later.